app/src/androidTest/java/de/test/antennapod/storage/ExceptFavoriteCleanupAlgorithmTest.java

```diff
@@ -0,0 +1,91 @@
+package de.test.antennapod.storage;
+
+import androidx.test.filters.SmallTest;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.storage.DBTasks;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests that the APFavoriteCleanupAlgorithm is working correctly.
+ */
+@SmallTest
```

Comment:
Could you please rebase the PR and migrate this to Roboelectric? See #4820 for details.